metrics/sanitize.go

```diff
@@ -0,0 +1,47 @@
+package metrics
+
+import (
+	"strings"
+	"unicode"
+)
+
+// SanitizeLabel sanitizes a label to conform with Prometheus label rules
+func SanitizeLabel(input string) string {
+	input = strings.TrimSpace(input)
+
+	if input == "" {
+		return input
+	}
+
+	var result strings.Builder
+
+	runeInput := []rune(input)
+	firstCh := runeInput[0]
+	// To prevent leading non-letter symbol
+	isAllowedLeadingCharacter := unicode.IsLetter(firstCh) || firstCh == '_'
+	if !isAllowedLeadingCharacter {
```

Comment:
I still do not get the logic:

why can't it be

```
isAllowedLeadingCharacter := unicode.IsLetter(firstCh)
if !isAllowedLeadingCharacter {
   result.WriteRune('_')
}
```

and that's all?

Reply:
the idea was to stay explicit, but perhaps you're right - simplified version is readable enough